src/app/(dashboard)/utils/useSiwe.ts

```diff
@@ -0,0 +1,57 @@
+import axios from "axios";
+import { JsonRpcSigner } from "ethers";
+import { SiweMessage } from "siwe";
+
+let domain: string;
+let origin: string;
+if (typeof window !== "undefined") {
+  domain = window.location.host;
+  origin = window.location.origin;
+}
+
+async function createSiweMessage(
+  address: string,
+  statement: string,
+  nonce: string,
+) {
+  if (!domain || !origin)
+    throw new Error("Missing config to create a SIWE message");
+  const expiry = new Date(Date.now());
+  expiry.setMinutes(expiry.getMinutes() + 5);
+  const expirationTime = expiry.toISOString();
+
+  const message = new SiweMessage({
+    domain,
+    address,
+    statement,
+    uri: origin,
+    version: "1",
+    chainId: 1,
```

Comment:
You should use the chainID from the wagmi hook here, so we can sign the message with the chain the user is currently using

Reply:
You can also get the chainId from the signer if I'm not mistaken